dataloader.go

```diff
@@ -20,18 +20,47 @@ import (
 // different access permissions and consider creating a new instance per
 // web request.
 type Interface interface {
-	Load(context.Context, interface{}) Thunk
-	LoadMany(context.Context, []interface{}) ThunkMany
-	Clear(context.Context, string) Interface
+	Load(context.Context, Keyer) Thunk
+	LoadMany(context.Context, KeyList) ThunkMany
+	Clear(context.Context, Keyer) Interface
 	ClearAll() Interface
-	Prime(ctx context.Context, key interface{}, value interface{}) Interface
+	Prime(ctx context.Context, key Keyer, value interface{}) Interface
+}
+
+// StringKey implements the Key interface for a string
+type StringKey string
+
+// Key is an identity method. Used to implement Key inteface
+func (k StringKey) Key() string { return string(k) }
+
+// String is an identity method. Used to implement Key Value
+func (k StringKey) Value() interface{} { return k }
+
+// Keyer is the interface that all keys need to implement
+type Keyer interface {
+	// Key returns a garunteed unique string that can be used to identify an object
+	Key() string
+	// Value returns the raw, underlaying value of the key
+	Value() interface{}
```

Comment:
`Value()` defined on `Keyer` threw me for a loop at first glance. It's a little strange that the `Keyer` points to a value, and has a `Value()` itself. 

Maybe naming this something like `Raw()` would avoid confusion? Not sure if that's unclear or un-idiomatic.